x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/details_page/details_page_filter_fields.tsx

```diff
@@ -154,6 +166,41 @@ export const MappingsFilter: React.FC<Props> = ({
             </div>
           )}
         </EuiSelectable>
+        <EuiPopoverFooter paddingSize="s">
+          <EuiButtonEmpty
+            color="danger"
+            iconSide="left"
+            size="s"
+            css={{ width: '100%' }}
+            iconType="cross"
+            data-test-subj="clearFilters"
+            disabled={isClearAllFilterDisabled}
+            onClick={() => {
+              if (!isAddingFields) {
+                state.filter.selectedOptions.filter((option) => {
+                  if (option.checked === 'on') {
+                    option.checked = undefined;
+                  }
+                });
+                setSelectedOptions(state.filter.selectedOptions);
+              } else {
+                previousState.filter.selectedOptions.filter((option) => {
+                  if (option.checked === 'on') {
+                    option.checked = undefined;
+                  }
+                });
+                setPreviousStateSelectedOptions(previousState.filter.selectedOptions);
+              }
+            }}
```

Comment:
I think this can be done like this to avoid repiting code:

```suggestion
            onClick={() => {
              if (!isAddingFields) {
                setSelectedOptions(clearChecked(state.filter.selectedOptions));
              } else {
                setPreviousStateSelectedOptions(clearChecked(previousState.filter.selectedOptions));
              }
            }}
```
and the `clearChecked` would be something like this:
```
  const clearChecked = (options: EuiSelectableOption[]) =>
    options.map((option) => (option.checked === 'on' ? { ...option, checked: undefined } : option));
```

Reply:
Sounds good. No strong opinion here. Added for more clear code. will make change

Reply:
updated in https://github.com/elastic/kibana/pull/221050/commits/e7fbfadafda61591c49cd4d707f8758f9ab3a98b